src/backend/termion.rs

```diff
@@ -137,11 +141,20 @@ where
     }
 
     fn get_cursor(&mut self) -> io::Result<(u16, u16)> {
-        termion::cursor::DetectCursorPos::cursor_pos(&mut self.writer).map(|(x, y)| (x - 1, y - 1))
+        match termion::cursor::DetectCursorPos::cursor_pos(&mut self.writer) {
+            Ok((x, y)) => {
+                self.cursor = Some((x - 1, y - 1));
+                Ok((x - 1, y - 1))
+            }
+            Err(_) => self
+                .cursor
+                .ok_or_else(|| io::Error::other("No last known cursor position")),
```

Comment:
Can a cursor position of 0,0 just be assumed as a fallback to the fallback? Or could that have issues?

Reply:
I think that could work. I discovered this issue with auto-resizing an inline viewport. Can't tell if that would have other side effects then the frame being redrawn at 0,0.

Reply:
This would means the underlying error will never be detectable. I think I'd prefer the error to just pass through here. Does that make sense?

I.e. if error and no fallback based on the previously set cursor, then just let the original error pass through